packages/lyra/src/prefix-tree/trie.ts

```diff
@@ -72,12 +75,24 @@ export class Trie {
         }
 
         if (!(word in _output)) {
-          _output[word] = new Set();
+          if (tolerance) {
+            // if the tolerance is set, we need to calculate the distance using levenshtein algorithm
+            distance = levenshtein(term, word);
```

Comment:
This looks perfect, I would just add a simple control like `if (word.length <= tolerance)` to avoid computing the edit distance on large strings, which will be ignored anyway. What do you think?

Reply:
I was thinking about it, I've added a check in the first commit that checks the **word.lenght**, but I wasn't completely convinced.

We can do something like this:

```js
if (tolerance) {
  // computing the difference of letters between the term and the word
  const difference = Math.abs(term.length - word.length);

  if (difference <= tolerance) {
    // if the tolerance is set, we need to calculate the distance using levenshtein algorithm
    distance = levenshtein(term, word);
...
```

Computing the difference of letters between the **word** and the **term** we can avoid on large strings.

It could be a good solution in your opinion?

Reply:
Yes, let's do this!